### PR TITLE
Fix merging multiple infoboxes

### PR DESCRIPTION
**Pull request by mubashar1199**
Solved the issue https://github.com/dbpedia/extraction-framework/issues/709, also added a new SHACL test and new page to minidump for testing

### CODE REVIEW

**Comment by jlareck** [comment]
@mubashar1199 please move `#Marian_Breland_Bailey,yes,yes` from `testGroups.csv` file to `shacl-test-groups.csv`. And delete the `dump/src/test/resources/testGroups.csv`, because for loading test groups we use `shacl-test-groups.csv` in dev branch

**Comment by mubashar1199** [comment]
> @mubashar1199 please move `#Marian_Breland_Bailey,yes,yes` from `testGroups.csv` file to `shacl-test-groups.csv`. And delete the `dump/src/test/resources/testGroups.csv`, because for loading test groups we use `shacl-test-groups.csv` in dev branch and `testGroups.csv` was deleted

Done, please see now

File: core/src/main/scala/org/dbpedia/extraction/mappings/TemplateMapping.scala [hunk 1]
```diff
@@ -6,180 +6,231 @@ import org.dbpedia.extraction.transform.Quad
...
+        // Condition #4
+        //if we have more than one info boxes and name property of infobox is different than title of page than infobox belong to different
+        //entity and need to be saved as new resource.
+        //This is to avoid assigning of one entity properties to another entity because a single page can have multiple info boxes and each
+        //about different entity, see dbr:Helene_Demuth
 
-        //Cannot generate URIs for empty templates
-        if(properties.isEmpty)
+        //checking if node is an infobox
+        var isInfobox = false
```
Comment (jlareck) [inline core/src/main/scala/org/dbpedia/extraction/mappings/TemplateMapping.scala:99]: You can also avoid making `isInfobox` mutable by rewriting in this way:
```
        val isInfobox = if (node.title.decoded.contains("Infobox")) {
          true
        } else {
          false
        }
```

File: core/src/main/scala/org/dbpedia/extraction/mappings/TemplateMapping.scala [hunk 2]
```diff
@@ -6,180 +6,231 @@ import org.dbpedia.extraction.transform.Quad
...
-    }
+  /**
+   * when extractor has a pre-phase
+   */
+
+  override def extract(node: TemplateNode, subjectUri: String): Seq[Quad] =
+  {
+    val pageNode = node.root
+    val graph = new ArrayBuffer[Quad]
+    logger.info("entity " + subjectUri)
```
Comment (jlareck) [inline core/src/main/scala/org/dbpedia/extraction/mappings/TemplateMapping.scala:41]: At the moment logging is a global issue in the Extraction Framework, and as I checked on my machine this line: `logger.info("entity " + subjectUri)` doesn't produce any message to the console or to any file (possibly I missed somewhere it). But anyway I also think that there is no need for this message during the extraction, so please remove it (also remove `private val logger = Logger.getLogger(TemplateMapping.getClass.getName)` in the line 30)

**Comment by jlareck** [comment]
@mubashar1199 PR is good, but there are some places where you just need to clean the code a little bit (I wrote the comments where you should do it)

File: core/src/main/scala/org/dbpedia/extraction/mappings/TemplateMapping.scala [hunk 3]
```diff
@@ -6,180 +6,231 @@ import org.dbpedia.extraction.transform.Quad
...
-        node.setAnnotation(TemplateMapping.INSTANCE_URI_ANNOTATION, uri)
+        //Extract properties
+        graph ++= mappings.flatMap(_.extract(node, subjectUri))
+        /* val allnames = node.children.filter(p => p.key == "name")
+         for (elem <- allnames)
+             name = elem.propertyNodeValueToPlainText;
+                 */
+      }
+      case Some(pageClass) => //This page already has a root template.
+      {
```
[the remark this answers is not in the document]
Comment (mubashar1199) [reply core/src/main/scala/org/dbpedia/extraction/mappings/TemplateMapping.scala]: resolved

File: core/src/main/scala/org/dbpedia/extraction/mappings/TemplateMapping.scala [hunk 4]
```diff
@@ -6,180 +6,231 @@ import org.dbpedia.extraction.transform.Quad
...
+        if(node.title.decoded.contains("Infobox"))
+          isInfobox = true
+
+        var condition4_SameEntityInfoBox = true
+
+        if(isInfobox)
         {
-            return templateNode.generateUri(subjectUri, templateNode.title.decoded)
+          //getting name property from infobox
+          val allnames = node.children.filter(p => p.key == "name")
```
[the remark this answers is not in the document]
Comment (mubashar1199) [reply core/src/main/scala/org/dbpedia/extraction/mappings/TemplateMapping.scala]: resolved

File: core/src/main/scala/org/dbpedia/extraction/mappings/TemplateMapping.scala (hunk 1 above)
Comment (mubashar1199) [reply core/src/main/scala/org/dbpedia/extraction/mappings/TemplateMapping.scala]: resolved

File: core/src/main/scala/org/dbpedia/extraction/mappings/TemplateMapping.scala (hunk 2 above)
Comment (mubashar1199) [reply core/src/main/scala/org/dbpedia/extraction/mappings/TemplateMapping.scala]: resolved

**Comment by mubashar1199** [comment]
> @mubashar1199 PR is good, but there are some places where you just need to clean the code a little bit (I wrote the comments where you should do it)

Thanks, done.

**Review by jlareck — COMMENTED** [review]
Great job! PR is cool but you still need to make some small changes

File: dump/src/test/resources/ci-tests/dbpedia-specific-ci-tests.ttl [hunk 5]
```diff
@@ -298,18 +298,3 @@ validator:itsrdf
 		v:pattern "^((?!(http://(ga\\.|af\\.|als\\.|am\\.|an\\.|ar\\.|arz\\.|ast\\.|az\\.|azb\\.|ba\\.|bar\\.|batsmg\\.|be\\.|bg\\.|bn\\.|bpy\\.|br\\.|bs\\.|bug\\.|ca\\.|cdo\\.|ce\\.|ceb\\.|ckb\\.|cs\\.|cv\\.|cy\\.|da\\.|de\\.|el\\.|eml\\.|en\\.|eo\\.|es\\.|et\\.|eu\\.|fa\\.|fi\\.|fo\\.|fr\\.|fy\\.|gd\\.|gl\\.|gu\\.|he\\.|hi\\.|hr\\.|hsb\\.|ht\\.|hu\\.|hy\\.|ia\\.|id\\.|ilo\\.|io\\.|is\\.|it\\.|ja\\.|jv\\.|ka\\.|kk\\.|kn\\.|ko\\.|ku\\.|ky\\.|la\\.|lb\\.|li\\.|lmo\\.|lt\\.|lv\\.|mai\\.|mg\\.|mhr\\.|min\\.|mk\\.|ml\\.|mn\\.|mr\\.|mrj\\.|ms\\.|my\\.|mzn\\.|nan\\.|nap\\.|nds\\.|ne\\.|new\\.|nl\\.|nn\\.|no\\.|oc\\.|or\\.|os\\.|pa\\.|pl\\.|pms\\.|pnb\\.|pt\\.|qu\\.|ro\\.|ru\\.|sa\\.|sah\\.|scn\\.|sco\\.|sd\\.|sh\\.|si\\.|simple\\.|sk\\.|sl\\.|sq\\.|sr\\.|su\\.|sv\\.|sw\\.|ta\\.|te\\.|tg\\.|th\\.|tl\\.|tr\\.|tt\\.|uk\\.|ur\\.|uz\\.|vec\\.|vi\\.|vo\\.|wa\\.|war\\.|wuu\\.|xmf\\.|yi\\.|yo\\.|yue\\.|zh\\.)?dbpedia.org/resource/.*/.)).)*$"
 		] .
```
[the remark this answers is not in the document]
Comment (mubashar1199) [reply dump/src/test/resources/ci-tests/dbpedia-specific-ci-tests.ttl]: resolved.

File: dump/src/test/resources/cv-test-groups.csv [hunk 6]
```diff
@@ -14,5 +14,3 @@ TEST_NAME,ALL,PRODUCTIVE
 #wikidata_IRIs,yes,yes
 #abstracts,yes,yes
 #forward_slash_in_resource_names,yes,yes
```
[the remark this answers is not in the document]
Comment (mubashar1199) [reply dump/src/test/resources/cv-test-groups.csv:16]: resolved

**Comment by mubashar1199** [comment]
> Great job! PR is cool but you still need to make some small changes

Thank you, both changes are made.

**Comment by mubashar1199** [comment]
@jlareck Thanks for merging, can you provide me any contribution letter/certificate from DBpedia for research purpose?

**Comment by jlareck** [comment]
@JJ-Author @Vehnem is it possible to provide any contribution letter for @mubashar1199 from DBpedia?

**Comment by JJ-Author** [comment]
@mubashar1199 please send a mail to dbpedia@infai.org

**Comment by mubashar1199** [comment]
> @mubashar1199 please send a mail to [dbpedia@infai.org](mailto:dbpedia@infai.org)

Sent few days ago, but no response yet

**Pull request merged**